Keep track of process instances to delete in RDBMS

konstantinos-mylosis opened the pull request:
## Description

The `HistoryDeletionIntent.DELETED` event is handled and the process instance to delete is stored in RDBMS.
Also created the get and delete methods

We may need to update these once they’re actually in use. 
e.g. on search add filtering/sorting mechanism, if necessary
I tried to follow the same approach as the ES/OS retrieval and removal.

## Related issues

closes #41529

konstantinos-mylosis committed:
feat: create history deletion table

add HistoryDeletionDbModel

konstantinos-mylosis committed:
feat: add history deletion mapper

konstantinos-mylosis committed:
feat: add history deletion writer

konstantinos-mylosis committed:
feat: add history deletion deleted handler

konstantinos-mylosis committed:
test: history deletion writer

konstantinos-mylosis committed:
feat: add history deletion reader

konstantinos-mylosis committed:
test: history deletion reader mock

konstantinos-mylosis committed:
test: history deletion IT

test insert, get, delete

konstantinos-mylosis committed:
test: enhance list of all known RDBMS table names

fix PurgerCompletenessIT

konstantinos-mylosis committed:
test: history deletion deleted handler

remcowesterhoud commented:
At the same time, maybe we need to create an entity in the `camunda-search-domain`? It appears this is what pretty much all other entities do. Eg, `JobEntity` with the `JobState` and `JobKind`

konstantinos-mylosis committed:
fix: history deletion rdbms

remove partitionId index
add DbModel enum, instead of using zeebe protocol one
fix oracle order by and limit query
add more test cases in HistoryDeletionIT

remcowesterhoud commented on qa/acceptance-tests/src/test/java/io/camunda/it/rdbms/db/historydeletion/HistoryDeletionIT.java:
```diff
@@ -0,0 +1,199 @@
+    historyDeletionWriter.delete(3, 9);
+    historyDeletionWriter.delete(RESOURCE_KEY, BATCH_OPERATION_KEY);
+    rdbmsWriter.flush();
+  }
```
🔧  It's probably easier to just purge using the purge mapper in the `rdbmsWriter`

remcowesterhoud reviewed (approved):
Just a few minor points left. Pre-approving.
